### PR TITLE
Add door_codes table and model

### PR DESCRIPTION
**Pull request by anaulin**
This PR depends on https://github.com/doubleunion/arooo/pull/441 , and can't be merged until that change is merged.

### What does this code do, and why?

This PR adds a a `DoorCode` model. This is the first step in the work for https://github.com/doubleunion/arooo/issues/360

### How is this code tested?

See specs.

### Are any database migrations required by this change?

Yes! That's the point!

### Screenshots (before/after)

No app or UI changes yet, screenshots not relevant.

### Are there any configuration or environment changes needed?
 
Nope.

### CODE REVIEW

File: app/models/door_code.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# == Schema Information
+#
+# Table name: door_codes
+#
+#  id         :integer          not null, primary key
+#  code       :string           not null
+#  enabled    :boolean          default(FALSE), not null
+#  created_at :datetime
+#  updated_at :datetime
+#  user_id    :integer
```
Comment (compwron) [inline app/models/door_code.rb:17]: I think `user_id` this should be a not null foreign key. It isn't showing that way in this autogenerated comment, which makes me worry that it is not.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -121,4 +121,15 @@
...
     end
   end
+
+  describe "#door_code" do
+    subject { create(:user) }
+
+    it { is_expected.to have_one(:door_code) }
+
+    it "defaults to nil" do
+      new_user = User.create!(name: "Kay Doke", email: "k@ok.com", username: "k")
```
Comment (compwron) [inline spec/models/user_spec.rb:131]: http://ok.com/ is a real site, I would prefer to use http://example.com/ even in tests where it doesn't matter 😅

**Comment by anaulin** [comment]
Thanks for the thoughtful comments. I have addressed them by:
* Made `user_id` foreign key explicitly non-null in the database schema. Also added a Rails model-level validation for it to be present.
* Added an index to `door_codes` table over `user_id`. This index also enforces that the `user_id` is unique, which means only one `DoorCode` can exist per `User` (we never discussed this axis of uniqueness, but it seems to make sense to me). I wasn't sure what index were you referring to (maybe you forgot to send that comment?), but looking at the schema it seems we'll be looking up codes by user at some point, so that made sense.
* Fixed test to use `@example.com` email.

@compwron can you make a second review pass on this, when you get a chance? 🙏

**Comment by anaulin** [comment]
This change is now deployed to `staging` and `production`.

**Pull request merged**